src/hotspot/cpu/ppc/ppc.ad

```diff
@@ -6537,6 +6537,7 @@ instruct storeNKlass(memory dst, iRegN_P2N src) %{
 // Store Pointer
 instruct storeP(memoryAlg4 dst, iRegPsrc src) %{
```

Comment:
The ADLC peephole replacement logic is not barrier data-aware as far as I can see. I wonder if barrier data is propagated correctly by peephole rules like the one in https://github.com/openjdk/jdk/blob/17a3f0e2577f2f9eb3fe62a4b8261e3dbe4c3b28/src/hotspot/cpu/ppc/ppc.ad#L14877-L14881. But I guess this is a question for PPC maintainers.

Reply:
Indeed. @TheRealMDoerr would you mind having a look at this?

Reply:
`OptoPeephole` is a `develop_pd` flag which is only enabled on x86. So, we don't support it for PPC64.
Even if somebody switched it on for PPC64, I don't think it would break anything.
The predicates are used during the matching phase. The peephole optimizer runs later at a point of time at which the barrier data is no longer used.

Reply:
Ah, good. Thanks for checking.